core/libp2p/crypto/proto/keys.proto

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+syntax = "proto3";
+package libp2p.crypto.proto;
+
+enum KeyType {
+    kUnspecified = 0;
+    kRSA1024 = 1;
+    kRSA2048 = 2;
+    kRSA4096 = 3;
+    kED25519 = 4;
```

Comment:
in JS key types are 
```
enum KeyType {
  RSA = 0;
  Ed25519 = 1;
  Secp256k1 = 2;
}
```
I propose you to use same types.

And I propose you to use Unspecified as internal type (enum in C++, not in proto); it can represent an unknown type:
```
switch(type){
case Type1: ...
case Type2: ...
default: return Unspecified;
}
```

Reply:
refactored